src/Plot.tsx

```diff
@@ -22,41 +24,56 @@ interface State {
 }
 
 function reducer(state: State, action: ReducerActions): State {
-  switch (action.type) {
-    case 'newData': {
-      const { series } = state;
-      const serieItem = action.value;
-      return { ...state, series: [...series, serieItem] };
+  return produce(state, (draft) => {
```

Comment:
instead of always returning `produce()` you can wrap the reducer function with `produce(function reducer(){})`